### ISSUE TITLE
ExternalCommunicator.cs freezing build when GetCommand() is called from Academy.cs Awake() method

### ISSUE THREAD

**Issue by MDReptile** [Unity-Technologies/ml-agents#201]
Unity 2017.3.0f3, Windows 7 64 bit, and following along with this tutorial series:

https://www.youtube.com/watch?v=-YWts_GlHCY&list=PLB5_EOMkLx_Ub1A4iHoDUx7vg37sVoL-E

I ran into an issue where as I try to run the ball example for training, and from the PPO page I try to run the step to "train the agents" the application opens in a small window, but it is all white, seemingly frozen.

I used debug.log statements through the code to find where it seems to freeze, and it makes it as far as the GetCommand() method in ExternalCommunicator.cs, and stops responding.

**Comment by MDReptile** [comment]
This could have something to do with Nvidia optimus failing to use the Nvidia GPU over the Intel GPU - as unity logs show the Intel GPU is being selected.

**Comment by awjuliani** [comment]
Hi @MDReptile, can you try launching `UnityEnvironment` in python using a different port?

**Comment by MDReptile** [comment]
I tried several ports, it seems after it freezes, the port I used is still held open, and it complains saying it cannot connect, so I used several different ports and would switch between them as I tested why it was crashing.

**Comment by MDReptile** [comment]
Ok - I think I've made some progress on this issue. I was concerned the optimus on my machine isn't working, and I always saw the intel card as being selected by unity in the log outputs.

So what I've tried to do, is using a batch file to temporarily disable, then re-enable to intel card, I've been able to get unity to select the nvidia card.

This causes another problem, where the resizing of the window causes the game to crash (it can't resize because the screen is blank, as the intel card is the buffer for the nvidia card). I think I may be able to get it working if I can prevent the resizing of the screen through code, so I'll report back what happens with further testing.

**Comment by awjuliani** [comment]
Thanks for exploring this and reporting your results, @MDReptile. We don't have access to a dual intel/nvidia laptop on our team, so we haven't gotten a chance to test it with that set-up. If you happen to figure out how to ensure it works under those conditions, we can share them on the documentation to help out others with the same set-up.

**Comment by MDReptile** [comment]
I've gotten to the bottom of this.

Turns out, my notebooks nvidia GPU isn't even capable of doing this :)

I have a video card that supports compute 2.1 version - which unfortunately is too old, as they require 3.0 or higher compatibility for tensorflow-gpu.

This means I'm limited to doing CPU based training only - and I discovered that when I tried this out on a desktop which also has a similar video card, and also only supports compute 2.1, which means I'll need to upgrade that desktops video card in order to use GPU compute for training the agents.

A list of video cards and what versions they support (for users facing this problem):
https://developer.nvidia.com/cuda-gpus

This issue can be closed.
